helper.php

```diff
@@ -394,31 +334,40 @@ public function tpl_tags($print = true) {
     /**
      * @param string $namespace empty for entire wiki
      *
+     * @param string $order_by
+     * @param bool $desc
+     * @param array $filters
      * @return array
      */
-    public function getAllTags($namespace = '', $order_by = 'tag', $desc = false) {
-        $order_fields = array('pid', 'tid', 'orig', 'taggers', 'count');
+    public function getAllTags($namespace = '', $order_by = 'tag', $desc = false, $filters = []) {
+        $order_fields = array('pid', 'tid', 'orig', 'taggers', 'ns', 'count');
         if (!in_array($order_by, $order_fields)) {
             msg('cannot sort by ' . $order_by . ' field does not exists', -1);
             $order_by = 'tag';
         }
 
+        list($having, $params) = $this->getFilterSql($filters);
+
         $db = $this->getDb();
 
         $query = 'SELECT    "pid",
                             CLEANTAG("tag") AS "tid",
                             GROUP_SORT(GROUP_CONCAT("tag"), \', \') AS "orig",
                             GROUP_SORT(GROUP_CONCAT("tagger"), \', \') AS "taggers",
+                            GROUP_SORT(GROUP_CONCAT(GET_NS("pid")), \', \') AS "ns",
+                            GROUP_SORT(GROUP_CONCAT("pid"), \', \') AS "pids",
```

Comment:
I don't like that we query all pages for all tags here. This should be requested on a per tag basis when needed. I would also argue that the `ns` info is not always needed.

I think this should be transformed to use the querybuilder as well.

Reply:
This method is used only in tag management (overview table). We actually do need namespaces here. As to pages: we need those too, because there is an additional ACL check for write access to pages when rendering the action icons. I'm not quite sure how it could be done better.

Reply:
Regarding the write access, as far as I understand, we only need the one information that at least one of the pages a tag is used on is writable by the user. And we only need this information when the user is not the super user.

So would it work to select something like `MAX(GETACCESSLEVEL(pid))` when querying for users and select a fixed `255` when querying for superusers?